go/mcap/writer.go

```diff
@@ -607,10 +607,21 @@ func (w *Writer) Close() error {
 			}
 		}
 	}
+	var summaryCrc uint32
+	if summarySectionStart != 0 && w.opts.IncludeCRC {
+		// add partial footer record to end of summary section CRC
+		footerPrefix := make([]byte, 1+8+8+8)
+		footerPrefix[0] = byte(OpFooter)
+		putUint64(footerPrefix[1:], 8+8+4)
+		putUint64(footerPrefix[1+8:], summarySectionStart)
+		putUint64(footerPrefix[1+8+8:], summaryOffsetStart)
+		w.w.w.crc.Write(footerPrefix)
```

Comment:
this is accessible because the module has a flat structure, but it shouldn't really be public to internal consumers. It would make sense to further split up the module to make this kind of access illegal.